<?xml version="1.0" encoding="UTF-8"?> 
<launch>
    <!-- Parameters -->
    <arg name="use_sim_time"              default="false" />
    <param name="use_sim_time"            value="$(arg use_sim_time)"/>

    <!-- GMSF Input Topics -->
    <arg name="imu_topic_name"            default="/sensors/imu" />
    <arg name="lidar_odometry_topic_name" default="/compslam_lio/odom_aft_mapped_to_init_CORRECTED" />
    <arg name="gnss_topic_name"           default="/rtk_gps_driver/position_receiver_0/ros/navsatfix" />

    <!-- Servers -->
    <arg name="gnss_coordinates_to_enu_topic_name"  default="/dual_graph_node/get_path_in_enu" />

    <!-- Config -->
    <arg name="extrinsics_from_urdf"      default="false" />
    <arg name="using_gnss"                default="false" />

    <!-- Config Files -->
    <arg name="core_graph_config_param_file"     default="$(find anymal_estimator_graph)/config/core/core_graph_config.yaml"/>
    <arg name="core_graph_param_file"            default="$(find anymal_estimator_graph)/config/core/core_graph_params.yaml"/>
    <arg name="core_extrinsic_param_file"        default="$(find anymal_estimator_graph)/config/core/core_extrinsic_params.yaml"/>
    <arg name="anymal_graph_param_file"          default="$(find anymal_estimator_graph)/config/anymal_specific/anymal_graph_params.yaml"/>
    <arg name="anymal_extrinsic_param_file"      default="$(find anymal_estimator_graph)/config/anymal_specific/anymal_extrinsic_params.yaml"/>
    <arg name="anymal_gnss_param_file"           default="$(find anymal_estimator_graph)/config/anymal_specific/anymal_gnss_params.yaml"/>
    <arg name="anymal_traj_alignment_param_file" default="$(find anymal_estimator_graph)/config/anymal_specific/anymal_traj_align_params.yaml"/>

    <!-- Node -->
    <node pkg="anymal_estimator_graph" type="anymal_estimator_graph_node" name="anymal_estimator_node" output="screen" > <!--launch-prefix="gdb -ex run -args"-->
        <!-- Launch Parameters -->
        <param name="launch/usingGnss" type="bool" value="$(arg using_gnss)" />
        <!-- Parameter files -->
        <rosparam command="load" file="$(arg core_graph_config_param_file)" />
        <rosparam command="load" file="$(arg core_graph_param_file)" />
        <rosparam command="load" file="$(arg core_extrinsic_param_file)" />
        <rosparam command="load" file="$(arg anymal_graph_param_file)" />
        <rosparam command="load" file="$(arg anymal_extrinsic_param_file)" />
        <rosparam command="load" file="$(arg anymal_gnss_param_file)" />
        <rosparam command="load" file="$(arg anymal_traj_alignment_param_file)" />
        <!-- Remapping of topics -->
        <remap from="/imu_topic"            to="$(arg imu_topic_name)" />
        <remap from="/lidar_odometry_topic" to="$(arg lidar_odometry_topic_name)" />
        <remap from="/gnss_topic"         to="$(arg gnss_topic_name)" />
        <!-- Remapping of services -->
        <remap from="/gnss_coordinates_to_enu_topic" to="$(arg gnss_coordinates_to_enu_topic_name)" />
    </node>

    <!-- TF for Aligning Compslam Map -->
    <node pkg="tf" type="static_transform_publisher" name="compslam_map_to_gmsf_map" args="0 0 0 0 0 0 map_graph_msf compslam_lio/camera_init_CORRECTED 10" />

</launch>
